app/javascript/components/Discover/Layout.tsx

```diff
@@ -47,60 +135,75 @@ export const Layout: React.FC<{
       : Routes.discover_url({ host: discoverDomain, taxonomy: newTaxonomyPath });
   };
 
-  const headerCta = currentSeller && (
-    <a href={Routes.library_url()} className="button">
-      <Icon name="bookmark-heart-fill" /> Library
+  const userActionButtons = <UserActionButtons />;
+
+  const logoLink = (
+    <a href={Routes.discover_path()} className="flex flex-shrink-0 items-center">
+      <img src={logo} alt="Gumroad" className="h-7 md:h-8 dark:invert" />
     </a>
   );
-
-  const avatar = currentSeller ? (
-    <a href={Routes.settings_main_url()} aria-label="Settings">
+  const searchBar = (
+    <div className="min-w-0 flex-grow">
+      <Search query={query} setQuery={setQuery} />
+    </div>
+  );
+  const cartButton = <CartNavigationButton className="link-button flex-shrink-0" />;
+  const avatarElement = currentSeller ? (
+    <a href={Routes.dashboard_url({ host: appDomain })} aria-label="Dashboard" className="flex-shrink-0">
       <img className="user-avatar" src={currentSeller.avatarUrl} />
     </a>
   ) : null;
 
-  const nav = (
+  const navElementWithFooter = (
+    <Nav
+      wholeTaxonomy={taxonomiesForNav}
+      currentTaxonomyPath={taxonomyPath}
+      onClickTaxonomy={onTaxonomyChange}
+      forceDomain={forceDomain}
+      footer={<div className="flex gap-4 border-b p-4 pb-4">{userActionButtons}</div>}
+    />
+  );
+
+  const navElementNoFooter = (
     <Nav
       wholeTaxonomy={taxonomiesForNav}
       currentTaxonomyPath={taxonomyPath}
       onClickTaxonomy={onTaxonomyChange}
       forceDomain={forceDomain}
-      footer={<footer>{headerCta}</footer>}
+      footer={undefined}
     />
   );
 
+  const headerRowElementsProps: HeaderRowElementsProps = {
+    logoLink,
+    searchBar,
+    userActionButtons,
+    cartButton,
+    avatarElement,
+    navElementNoFooter,
+    navElementWithFooter,
+  };
+
   return (
     <main className={cx("discover", className)}>
-      <section className="content sticky top-0 z-50 p-0">
-        <HomeNav />
-      </section>
       <header
         className="hero border-t-0 lg:pe-16 lg:ps-16"
         style={showTaxonomy && rootTaxonomy ? getRootTaxonomyCss(rootTaxonomy) : undefined}
       >
-        <div className="hero-actions">
-          <CartNavigationButton className="link-button" />
```

Comment:
@jc26 _a I think the cart button is missing now - passing this back to you for now and i'll fix the the tests laters